src/EditorFeatures/Core/Implementation/Classification/SemanticClassificationUtilities.cs

```diff
@@ -149,5 +209,210 @@ internal static class SemanticClassificationUtilities
                 throw ExceptionUtilities.Unreachable;
             }
         }
+
+        private async Task AddSemanticClassificationsAsync(
+            IClassificationService classificationService,
+            Document document,
+            TextSpan textSpan,
+            List<ClassifiedSpan> classifiedSpans,
+            CancellationToken cancellationToken)
+        {
+            var workspace = document.Project.Solution.Workspace;
+            var fullyLoadedStateTask = s_workspaceToFullyLoadedStateTask.GetValue(
+                workspace, w =>
+                {
+                    var workspaceLoadedService = workspace.Services.GetRequiredService<IWorkspaceStatusService>();
+                    return workspaceLoadedService.WaitUntilFullyLoadedAsync(CancellationToken.None);
+                });
+
+            // If we're not fully loaded try to read from the cache instead so that classifications appear up to date.
+            // New code will not be semantically classified, but will eventually when the project fully loads.
+            var isFullyLoaded = fullyLoadedStateTask.IsCompleted;
```

Comment:
Not just using https://github.com/dotnet/roslyn/blob/d21d2465cb126e6163c63d18d8bc664ce04a6d90/src/Workspaces/Core/Portable/Workspace/Host/Status/IWorkspaceStatusService.cs#L44?